src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -155,20 +130,20 @@ def get_all_updates(self, cached=False):
             self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
             return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
 
-        out = self.invoke_package_manager(self.tdnf_check)
+        out = self.invoke_package_manager(self.__generate_command(self.tdnf_check, self.max_patch_publish_date))
         self.all_updates_cached, self.all_update_versions_cached = self.extract_packages_and_versions(out)
-
         self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
         return self.all_updates_cached, self.all_update_versions_cached
 
     def get_security_updates(self):
         """Get missing security updates. NOTE: Classification based categorization of patches is not available in Azure Linux as of now"""
-        self.composite_logger.log_verbose("[TDNF] Discovering 'security' packages...")
-        security_packages, security_package_versions = [], []
+        self.composite_logger.log_verbose("[TDNF] Discovering all packages as 'security' packages, since Azure Linux does not support package classification...")
```

Comment:
TDNF does not support security classifications.
It's good to not be overly Azure Linux coupled in a generic Tdnf class from the get go even if the first consumer is Azure Linux.

Reply:
I'd actually take a closer look at the YumPackageManager implementation for a blueprint on how to think about this.

Reply:
Done